newIDE/app/src/AssetStore/AssetStoreNavigator.js

```diff
@@ -91,13 +91,21 @@ export const useNavigation = (): NavigationState => {
       getCurrentPage: () => previousPages[previousPages.length - 1],
       backToPreviousPage: () => {
         if (previousPages.length > 1) {
+          const newPreviousPages = previousPages.slice(
+            0,
+            previousPages.length - 1
+          );
+          const newCurrentPage = newPreviousPages[newPreviousPages.length - 1];
           setHistory({
             previousPages: previousPages.slice(0, previousPages.length - 1),
           });
+          return newCurrentPage;
         }
+        return previousPages[0];
```

Comment:
Should there be only one `return` to make it clear the the current page is always returned or maybe let callers use `getCurrentPage`?

Reply:
The problem is that React guarantees that the state of a component is the same only from within the same callback/effect/memo.
In the asset store, we call a method to navigate (that changes the current page) and then we call a callback to set the fact that we need a scroll update.
But this callback does not have an up-to-date state: when it gets the current page, it gets the previous current page.